src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -474,38 +475,31 @@ def handleJob(self, jobID, fwkJobReport):
             elif jobType == "Express" and set(outputMap).difference(outputModules) == {'write_RAWRAW'}:
                 pass
             else:
-                failJob = True
-                if jobType in ["Processing", "Production"]:
-                    cmsRunSteps = 0
-                    for step in fwkJobReport.listSteps():
-                        if step.startswith("cmsRun"):
-                            cmsRunSteps += 1
-                    if cmsRunSteps > 1:
-                        failJob = False
-
-                if failJob:
+                # Production and Processing filesets do not consider the logArchive step
+                if set(outputMap) != (outputModules - {"logArchive"}):
                     jobSuccess = False
-                    logging.error("Job %d , list of expected outputModules does not match job report, failing job",
-                                  jobID)
-                    logging.debug("Job %d , expected outputModules %s", jobID, sorted(outputMap.keys()))
-                    logging.debug("Job %d , fwjr outputModules %s", jobID, sorted(outputModules))
+                    errMsg = f"Job {jobID}, expected output modules: {sorted(outputMap.keys())}, "
```

Comment:
Honestly I did not get use to f-string syntax and seeing functions execution within seems weird. I rather prefer to get `sorted` function out of f-string, assign it to new variable and pass it to f-string, e.g.
```
keys = sorted(outputMap.keys())
errMsg = f".... {keys}"
```

Reply:
I find it not very friendly as well, other than being handy in reducing the line length. However, the implementation is totally valid and I'd suggest you to have a look at the documentation:
https://docs.python.org/3.8/tutorial/inputoutput.html#formatted-string-literals

I can make the first line more clear though to avoid possible mistakes in the future. Will squash with the 3rd commit.